Exercise_4.java

```diff
@@ -35,6 +40,32 @@ static void insert(Node temp, int key)
 
         // Do level order traversal until we find 
         // an empty place and add the node.  
+    	Node newNode = new Node(key);

```

Comment:
Good Job! Please add space and time complexity. Also, please review inline comments on previous problems.